railties/lib/rails/application/finisher.rb

```diff
@@ -124,6 +124,7 @@ def self.complete(_state)
       # the hook are taken into account.
       initializer :set_routes_reloader_hook do |app|
         reloader = routes_reloader
+        reloader.eager_load = app.config.eager_load
```

Comment:
Not sure what would it improve. Railtie also have `eager_load_namespaces` that automatically call `eager_load!` on them. But since the routes_reloader is created in an initializer we can't use it.